### ISSUE TITLE
Sails development is outpacing documentation

### ISSUE THREAD

**Issue by newz2000** [balderdashy/sails#2325]
Hello,

Sails.js is a useful framework that opens up a lot of doors. However, the learning curve is pretty steep. You can get the ultra basic stuff done well based on the tutorial, however going beyond "Hello World" is an exercise left to the end user alone.

The reach of Sails is limited to people with either very basic needs or to people who are already pretty strong with Node.js web development. Good documentation would make it easier to use the more powerful and complicated features.

I realize that Sails started as the scratch to an itch specifically felt by Balderdash. If that's all Sails is, then nothing needs to change. However, I think you've created something that will help the development world.

It also benefits the development team dramatically. Right now there are a relatively small pool of advanced users. I know this because I've been to the Google Group, the IRC channel and Stack Overflow. There just aren't that many people answering questions. The implication is that there aren't that many people with both the knowledge and time to help in that way.

By focusing on the documentation you can increase the number of intermediate developers using more of the framework. By making it easy to go from basic user to basic+ to intermediate to intermediate+ to expert you strengthen the community.

I look at some of the strongest development tools that have broad user basis and I see projects that had or have great documentation. PHP is probably the most infamous, but jQuery, Rails and Django are other great examples. In each of these cases, great documentation lowered the barrier to entry and boosted adoption.

* Rails has more than just API documentation, they have guides.
* Django has an excellent tutorial (though I do take some exception to their "hard way first" method of explanation)
* PHP provided a way for users to contribute to the documentation through a commenting system

If creating a list of topics that need to be documented, let me know and I can work on this, but here are a few examples:

* In the [Getting Started](http://sailsjs.org/#/getStarted) doc, there are a lot of TODOs.
* Every item in the Features section should be a link to an explanation of how to utilize this capability. For example, 
  * "Websockets" is a feature but it's not clear how to use this. 
  * Another great feature is "Policies" but it too lacks enough documentation to put it to use practically - specifically in order to use a policy you need to know something about a user but there is no explanation on how to do that.

I would like to help, and I am by doing some blogging and speaking on Sails at an upcoming conference, however I myself am at a loss on a lot of these topics and have not had great success in getting help when I get stuck. The big changes with the 0.10 release means that few of the tutorials on the web or Stack Exchange are relevant.

**Comment by sailsbot** [comment]
Thanks for posting, @newz2000.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
